src/components/ContentPreview/ContentPreview.js

```diff
@@ -237,28 +244,38 @@ class ContentPreview extends PureComponent<Props, State> {
      * @return {void}
      */
     componentDidMount(): void {
-        const { fileId }: Props = this.props;
-
         this.loadStylesheet();
         this.loadScript();
 
-        this.fetchFile(fileId);
+        this.fetchFile(this.state.currentFileId);
         this.focusPreview();
     }
 
+    static getDerivedStateFromProps(props: Props, state: State) {
+        const { fileId } = props;
+        if (state.prevFileId !== fileId && fileId !== state.currentFileId) {
```

Comment:
What if the incoming fileId is actually the previous file id. As in the parent actually set the prior one. In that case we have to cancel getting the current one and fetch the prior one again.